pgjdbc/src/main/java/org/postgresql/gss/GSSOutputStream.java

```diff
@@ -49,9 +49,9 @@ protected void flushBuffer() throws IOException {
 
   private void writeWrapped(byte[] b, int off, int len) throws IOException {
     try {
-      byte[] token = gssContext.wrap(buf, off, len, messageProp);
-      pgOut.writeInt4(token.length);
-      pgOut.write(token, 0, token.length);
+      byte[] token = gssContext.wrap(b, off, len, messageProp);
+      pgOut.writeInt4(len);
+      pgOut.write(token, 0, len);
```

Comment:
Why is this change needed?

Reply:
`b.length` may comes from a bigger buffer, `gssContext.wrap` will encode `b[off: off + len]` and return `b`. if `b` is bigger than `len`, `writeWrapped()` will wirte an incorrect length.

Reply:
Remember: we write to the `pgOut` stream, and we should write `length + contents`.
The contents is `token` byte array which is a **new** array, and it has nothing to do with `b`/`buf`.

Why did you replace `len`?

Reply:
ok....

gssContext.wrap() will not modify the source input